chainer_compiler/chainer_compiler.py

```diff
@@ -203,6 +203,34 @@ def _run_translator(translator, mc, inputs):
     return graph
 
 
+def _resolve_name_correspondence(mc, params, name, param_values):
+    for running_mean_name in ('avg_mean', 'running_mean'):
+        if not name.endswith(running_mean_name):
+            continue
+        parent = name[:-len(running_mean_name)]
+        mean = params[parent + 'beta']
+        for link in mc.links():
+            if not hasattr(link, 'avg_mean'):
+                continue
+            if any(id(p) == id(mean) for p in link.params()):
```

Comment:
I think `links.params` iterates the parameters recursively. If a user defines a chain with `avg_mean` defined, this function may misbehave.